### PR TITLE
Added unit test for getAccount()

### PR DESCRIPTION
**Pull request by bonniecpk**

### CODE REVIEW

File: oauthdoctor/oauth/helper.go [hunk 1]
```diff
@@ -255,17 +254,23 @@ func (c *Config) oauth2Client(code string) (*http.Client, string) {
...
+		return nil, err
+	}
+
 	req.Header.Set("developer-token", c.ConfigFile.DevToken)
 	if c.ConfigFile.LoginCustomerID != "" {
 		req.Header.Set("login-customer-id", c.ConfigFile.LoginCustomerID)
 	}
-	resp, err := client.Do(req)
+
+	var resp *http.Response
```
Comment (bobhancock) [inline oauthdoctor/oauth/helper.go]: Why is this declared as a var here? Why not:
resp, err = client.Do(req)

File: oauthdoctor/oauth/helper_test.go [hunk 2]
```diff
@@ -164,6 +166,74 @@ func TestReplaceRefreshToken(t *testing.T) {
...
+		{
+			desc: "developer-token is in HTTP header",
+			c: Config{
+				ConfigFile: diag.ConfigFile{
+					ConfigKeys: diag.ConfigKeys{
+						DevToken: "devToken",
+					},
+				},
+			},
+			ts: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```
Comment (bobhancock) [inline oauthdoctor/oauth/helper_test.go]: Nice.

File: oauthdoctor/oauth/helper.go [hunk 3]
```diff
@@ -255,17 +254,23 @@ func (c *Config) oauth2Client(code string) (*http.Client, string) {
 	return conf.Client(oauth2.NoContext, token), token.RefreshToken
 }
 
+var apiURL = "https://googleads.googleapis.com/v1/customers/"
```
Comment (bobhancock) [inline oauthdoctor/oauth/helper.go]: Why is there a default value if you are going to override it in the tests?

File: oauthdoctor/oauth/helper.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bonniecpk) [reply oauthdoctor/oauth/helper.go]: At this point, resp is undeclared and err is declared. Is there a neater way to do this?

File: oauthdoctor/oauth/helper.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (bonniecpk) [reply oauthdoctor/oauth/helper.go]: It will be modified in the test

Comment (bonniecpk) [reply oauthdoctor/oauth/helper.go]: The default value is needed for the actual Google Ads API call. It is declared here, so the unit test can override the URL from httptest server.

File: oauthdoctor/oauth/helper.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bobhancock) [reply oauthdoctor/oauth/helper.go]: You can use := if at least on variable on the left is undeclared.

File: oauthdoctor/oauth/helper_test.go [hunk 4]
```diff
@@ -164,6 +166,74 @@ func TestReplaceRefreshToken(t *testing.T) {
...
+			want: "This is an error",
+		},
+	}
+
+	for _, tt := range tests {
+		apiURL = tt.ts.URL
+		defer tt.ts.Close()
+
+		buf, err := tt.c.getAccount(tt.ts.Client())
+		if err != nil && errstring(err) != tt.want {
```
Comment (bobhancock) [inline oauthdoctor/oauth/helper_test.go]: Do you need the test for != nil? Aren't you testing for anything other than tt.want?

File: oauthdoctor/oauth/helper.go [hunk 5]
```diff
@@ -278,7 +282,7 @@ func (c *Config) getAccount(client *http.Client) (*bytes.Buffer, error) {
 	json.Unmarshal(buf.Bytes(), &jsonBody)
 
 	if jsonBody["error"] != nil {
-		return nil, errors.New(buf.String())
+		return nil, fmt.Errorf(jsonBody["error"].(string))
```
[the remark this answers is not in the document]
Comment (bonniecpk) [reply oauthdoctor/oauth/helper.go]: The returned err is used as an output message when users run the program, not for debugging purpose. Adding a function name will confuse the user who runs the program.

File: oauthdoctor/oauth/helper_test.go (hunk 4 above)
Comment (bonniecpk) [reply oauthdoctor/oauth/helper_test.go]: Will remove

Comment (bonniecpk) [reply oauthdoctor/oauth/helper_test.go]: I need it. When err is nil, I need to short-circuit the if-statement to skip `errstring(err) != tt.want`
